TradeItIosTicketSDK2/TradeItPortfolioBalanceFXPresenter.swift

```diff
@@ -1,30 +1,59 @@
 import TradeItIosEmsApi
 
-class TradeItPortfolioBalanceFXPresenter {
+class TradeItPortfolioBalanceFXPresenter: TradeItPortfolioBalancePresenter  {
 
-    let fxBalance: TradeItFxAccountOverview
-    let tradeItLinkedBrokerAccount: TradeItLinkedBrokerAccount
+    var fxBalance: TradeItFxAccountOverview = TradeItFxAccountOverview()
     
     init(_ tradeItLinkedBrokerAccount: TradeItLinkedBrokerAccount) {
-        self.tradeItLinkedBrokerAccount = tradeItLinkedBrokerAccount
-        self.fxBalance = tradeItLinkedBrokerAccount.fxBalance
+        if let fxBalance = tradeItLinkedBrokerAccount.fxBalance {
+            self.fxBalance = fxBalance
+        }
     }
     
     func getFormattedTotalValue() -> String {
-        return formatCurrency(self.fxBalance.totalValueUSD)
+        guard let totalValueUSD = self.fxBalance.totalValueUSD
+            else {return TradeItPresenter.MISSING_DATA_PLACEHOLDER}
+        return formatCurrency(totalValueUSD)
     }
     
     func getUnrealizedProfitAndLoss() -> String {
-        return formatCurrency(self.tradeItLinkedBrokerAccount.fxBalance.unrealizedProfitAndLossBaseCurrency)
+        guard let unrealizedProfitAndLossBaseCurrency = self.fxBalance.unrealizedProfitAndLossBaseCurrency
+            else {return TradeItPresenter.MISSING_DATA_PLACEHOLDER}
+        return formatCurrency(unrealizedProfitAndLossBaseCurrency)
     }
     
     func getRealizedProfitAndLoss() -> String {
-        return formatCurrency(self.tradeItLinkedBrokerAccount.fxBalance.realizedProfitAndLossBaseCurrency)
+        guard let realizedProfitAndLossBaseCurrency = self.fxBalance.realizedProfitAndLossBaseCurrency
+            else {return TradeItPresenter.MISSING_DATA_PLACEHOLDER}
+        return formatCurrency(realizedProfitAndLossBaseCurrency)
 
     }
     
     func getMarginBalance() -> String {
-        return formatCurrency(self.tradeItLinkedBrokerAccount.fxBalance.marginBalanceBaseCurrency)
+        guard let marginBalanceBaseCurrency = self.fxBalance.marginBalanceBaseCurrency
+            else {return TradeItPresenter.MISSING_DATA_PLACEHOLDER}
+        return formatCurrency(marginBalanceBaseCurrency)
+    }
+    
+    func getFormattedTotalValueWithPercentage() -> String{
+        guard let totalValueBaseCurrency = fxBalance.totalValueBaseCurrency
+            else { return TradeItPresenter.MISSING_DATA_PLACEHOLDER}
+        
+        var formattedTotalValue = NumberFormatter.formatCurrency(totalValueBaseCurrency)
+
+        let totalReturn = fxBalance.unrealizedProfitAndLossBaseCurrency?.floatValue
+        if (totalReturn != nil && totalReturn != 0) {
+            let totalPercentReturn = totalReturn! / (totalValueBaseCurrency.floatValue - abs(totalReturn!))
```

Comment:
This function only calculates the percentage, so it would be called by the other function you wrote.